servicecontrol/containerization/index.md

```diff
@@ -0,0 +1,171 @@
+
+ServiceControl can now be accessed over port `44444`.
+
+All [supported parameters](/servicecontrol/audit-instances/creating-config-file.md) for ServiceControl Audit can be specified via environment variables.
+
+### Monitoring Instances
+
+Monitoring instances can be run standalone and therefore only need the init container to be run and then the runtime container.
+
+`docker run -e "Monitoring/ConnectionString=[connectionstring]" -e 'Monitoring/LicenseText=[licensecontents]' -v c:/monitoringlogs/:c:/logs/ -d particular/servicecontrol.sqlserver.monitoring.init-windows:4`
+
+The init container will run and shut down once the required queues and database has been created.
+
+The runtime instance of ServiceControl Monitoring can now be run:
+
+`docker run -e "Monitoring/ConnectionString=[connectionstring]" -e 'Monitoring/LicenseText=[licensecontents]' -v c:/data/:c:/data/ -v c:/logs/:c:/logs/ -p 33633:33633 -d particular/servicecontrol.sqlserver.monitoring-windows:4`
+
+All [supported parameters](/servicecontrol/monitoring-instances/installation/creating-config-file.md) for ServiceControl Monitoring can be specified via environment variables.
+
+## Volume mappings
+
+A general practice with containers is that no data is persisted inside a container, but rather outside the container on either the host machine, database or something similar. This makes it easier to backup data, share data among containers and more importantly, not lose the data when the container is updated to a later version. Storing files outside of the container can be achieved with volume mappings, or with folder mappings for Windows containers.
+
+Volume mappings can be used with ServiceControl for configuration files, storing logfiles and possibly reading a license file. More information on data storage can be found on the [Docker website](https://docs.docker.com/storage/).
+
+### Configuration and Logging
+
+Every image has a folder in `c:\data\` that can be mapped to a folder outside of the container. 
+
+`docker run -v c:\servicecontrol\:c:\data\ [imagename]`
+
+As a result, that container will write both logfiles and its database into that folder in respectively the `Logs` and `DB` folders. Another option is to map each folder specifically, for more control over their locations and divide them over separate drives, for example:
+
+`docker run -v c:\servicecontroldatabase\:c:\data\DB\ -v d:\servicecontrollogfiles\:c:\data\logs\ [imagename]`
+
+### License file
+
+The license file can be provided via an environment variable on the command-line or via an environment file. Another options is to map the folder where ServiceControl will look for the license file.
+
+`docker run -v c:\servicecontrol\license\:%PROGRAMDATA%\ParticularSoftware\ [imagename]`
```

Comment:
This doesn't seem right:

The ENVVAR would be applied BEFORE the volume mapping. The right part is the container location. That is fixed to `C:\ProgramData`, the left part is the host.

Isn't this mapping failing if the host would have Program Data on D: ?

This is the mapping I use in the sample:

    ${PROGRAMDATA}/ParticularSoftware:C:/ProgramData/ParticularSoftware

Reply:
The host could never have it on D as they can't change where the container internally stores the data. We may as well hardcode the exact path here because it is not possible for anyone to change the container's location.